File: script-gen-ui/src/app/model/sweep_data/stepSweepConfig.ts
import { IStepGlobalParameters, ISweepGlobalParameters } from '../interface';
import { ParameterFloat, ParameterInt } from './SweepTimingConfig';

export class StepGlobalParameters {
  step_points: ParameterInt;
  step_to_sweep_delay: ParameterFloat;
  list_step: boolean;

  constructor(data: IStepGlobalParameters) {
    this.step_points = new ParameterInt(data.step_points);
    this.step_to_sweep_delay = new ParameterFloat(data.step_to_sweep_delay);
    this.list_step = data.list_step;
  }

  toJSON() {
    return {
      step_points: this.step_points.toJSON(),
      step_to_sweep_delay: this.step_to_sweep_delay.toJSON(),
      list_step: this.list_step,
    };
  }
}

export class SweepGlobalParameters {
  sweep_points: ParameterInt;
  list_sweep: boolean;

  constructor(data: ISweepGlobalParameters) {
    this.sweep_points = new ParameterInt(data.sweep_points);
    this.list_sweep = data.list_sweep;
  }

  toJSON() {
    return {
      sweep_points: this.sweep_points.toJSON(),
      list_sweep: this.list_sweep,
    };
  }
}

File: script-gen-manager/src/model/sweep_data/step_sweep_config.rs
use serde::{Deserialize, Serialize};
use std::str;

use crate::instr_metadata::base_metadata::BaseMetadata;

use super::parameters::{ParameterFloat, ParameterInt};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct StepGlobalParameters {
    pub step_points: ParameterInt,
    pub step_to_sweep_delay: ParameterFloat,
    pub list_step: bool,
}

impl StepGlobalParameters {
    #[must_use]
    pub fn new() -> Self {
        Self {
            step_points: ParameterInt::new("step_points", 10),
            step_to_sweep_delay: ParameterFloat::new(
                "step_to_sweep_delay",
                0.0,
                Some(BaseMetadata::UNIT_SECONDS.to_string()),
            ),
            list_step: false,
        }
    }
}

impl Default for StepGlobalParameters {
    fn default() -> Self {
        Self::new()
    }
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SweepGlobalParameters {
    pub sweep_points: ParameterInt,
    pub list_sweep: bool,
}

impl SweepGlobalParameters {
    #[must_use]
    pub fn new() -> Self {
        Self {
            sweep_points: ParameterInt::new("sweep_points", 10),
            list_sweep: false,
        }
    }
}

impl Default for SweepGlobalParameters {
    fn default() -> Self {
        Self::new()
    }
}
